proto/binlogdata.proto

```diff
@@ -200,6 +203,16 @@ enum OnDDLAction {
   EXEC_IGNORE = 3;
 }
 
+// VReplicationWorkflowType define types of vreplication workflows.
+enum VReplicationWorkflowType {
+  MATERIALIZE = 0;
+  MOVETABLES = 1;
+  CREATELOOKUPINDEX = 2;
+  MIGRATE = 3;
+  RESHARD = 4;
+  ONLINEDDL = 5;
```

Comment:
Worth adding MESSAGING here? I feel like we probably should.

Reply:
Delegating to #9742